services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocol.java

```diff
@@ -339,8 +341,30 @@ public Response backchannelLogout(UserSessionModel userSession, AuthenticatedCli
 
     @Override
     public Response frontchannelLogout(UserSessionModel userSession, AuthenticatedClientSessionModel clientSession) {
-        // todo oidc redirect support
-        throw new RuntimeException("NOT IMPLEMENTED");
+        if (clientSession != null) {
+            String url = clientSession.getClient().getAttribute("frontchannel.logout.url");
+            if (url != null && !url.trim().isEmpty()) {
+                UriBuilder builder = UriBuilder.fromUri(url);
+                builder.queryParam("sid", userSession.getId());
+                builder.queryParam("iss", clientSession.getNote(OIDCLoginProtocol.ISSUER));
+                URI uri = builder.build();
+                appendUserSession(userSession,"frontchannel_logout_urls",uri.toString());
```

Comment:
IMO, we should just pass a list of client ids. It should have the same effect as this because from the id you can query the client and then obtain the frontend uri.

Also, it should help building the freemarker template as we can get more information about the client (e.g.: display name, client id).

Reply:
Agree